Ginger/Ginger/UserControlsLib/UCDataMapping.xaml.cs

```diff
@@ -0,0 +1,553 @@
+                    xVariablesComboBox.Visibility = Visibility.Hidden;
+                }
+
+                if ((MappedType == eDataType.OutputVariable.ToString() || MappedType == eDataType.GlobalVariable.ToString() || MappedType == eDataType.ApplicationModelParameter.ToString())
+                    && xOptionalValuesComboBox != null)
+                {
+                    xOptionalValuesComboBox.Visibility = Visibility.Visible;
+                }
+                else
+                {
+                    xOptionalValuesComboBox.Visibility = Visibility.Hidden;
+                }
+
+                if (MappedType == eDataType.DataSource.ToString()
+                    && xDSExpressionTxtbox != null)
+                {
+                    xDSExpressionTxtbox.Visibility = Visibility.Visible;
+                    xDSConfigBtn.Visibility = Visibility.Visible;
+                }
+                else
+                {
+                    xDSExpressionTxtbox.Visibility = Visibility.Hidden;
+                    xDSConfigBtn.Visibility = Visibility.Hidden;
+                }
+            }
+        }
+
+        private static void OnMappedValuePropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            var control = sender as UCDataMapping;
+            if (control != null)
+            {
+                control.MappedValuePropertyChanged((string)args.NewValue);
+            }
+        }
+        private void MappedValuePropertyChanged(string mappedValueProperty)
+        {
+            if (MappedType == eDataType.OutputVariable.ToString()
+                || MappedType == eDataType.GlobalVariable.ToString()
+                || MappedType == eDataType.ApplicationModelParameter.ToString())
+            {
+                OnPropertyChanged(nameof(MappedValueGUID));
+            }
+            else
+            {
+                OnPropertyChanged(nameof(MappedValue));
+            }
+            
+            MarkMappedValueValidation();
+        }
+
+        private void MarkMappedValueValidation()
+        {
+            bool isValid = true;
+
+            if ((MappedType != eDataType.None.ToString() && MappedValue == string.Empty)
+                || (MappedType == eDataType.Variable.ToString() && !GingerCore.General.CheckComboItemExist(xVariablesComboBox, MappedValue))
+                || ((MappedType == eDataType.OutputVariable.ToString() || MappedType == eDataType.GlobalVariable.ToString() || MappedType == eDataType.ApplicationModelParameter.ToString()) && !GingerCore.General.CheckComboItemExist(xOptionalValuesComboBox, MappedValue, "Guid"))
+                || (MappedType == eDataType.DataSource.ToString() && GingerCoreNET.GeneralLib.General.CheckDataSource(MappedValue, WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<DataSourceBase>()) != string.Empty))               
+            {
+                isValid = false;
+            }
+
+            if (isValid == false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5538205)